kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/ClusterEventTracking.java

```diff
@@ -401,42 +401,38 @@ protected void writeBatchEvents(Collection<Event> events)
 		try
 		{
 			conn = sqlService().borrowConnection();
-			wasCommit = conn.getAutoCommit();
-			if (wasCommit)
-			{
-				conn.setAutoCommit(false);
-			}
 
-			// Note: investigate batch writing via the jdbc driver: make sure we can still use prepared statements (check out host arrays, too)
-			// -ggolden
+			// common preparation for each insert
+			String statement = insertStatement();
 
-            // common preparation for each insert
-            String statement = insertStatement();
-            Object fields[] = new Object[6];
+			// Setup a batch of events if not using a cluster
+			Set<Object[]> eventSet = new HashSet<>();
```

Comment:
This should probably be a LinkedHashSet to maintain order of items, right?

Reply:
Why not have the API use a `List` as in most cases order is always important isn't it?

Reply:
Yup, order can be important. I will make the change.